src/mlpack/methods/lars/lars.cpp

```diff
@@ -68,6 +68,120 @@ LARS::LARS(const arma::mat& data,
   Train(data, responses, transposeData);
 }
 
+// Copy Constructor.
+LARS::LARS(const LARS& other) :
+    matGramInternal(other.matGramInternal),
+    matGram(&matGramInternal),
+    matUtriCholFactor(other.matUtriCholFactor),
+    useCholesky(other.useCholesky),
+    lasso(other.lasso),
+    lambda1(other.lambda1),
+    lambda2(other.lambda2),
+    elasticNet(other.elasticNet),
+    tolerance(other.tolerance),
+    betaPath(other.betaPath),
+    lambdaPath(other.lambdaPath),
+    activeSet(other.activeSet),
+    isActive(other.isActive),
+    ignoreSet(other.ignoreSet),
+    isIgnored(other.isIgnored)
+{
+  // Nothing to do here.
+}
+
+// Move constructor.
+LARS::LARS(LARS&& other) :
+    matGramInternal(std::move(other.matGramInternal)),
+    matGram(other.matGram),
+    matUtriCholFactor(std::move(other.matUtriCholFactor)),
+    useCholesky(other.useCholesky),
+    lasso(other.lasso),
+    lambda1(other.lambda1),
+    lambda2(other.lambda2),
+    elasticNet(other.elasticNet),
+    tolerance(other.tolerance),
+    betaPath(std::move(other.betaPath)),
+    lambdaPath(std::move(other.lambdaPath)),
+    activeSet(std::move(other.activeSet)),
+    isActive(std::move(other.isActive)),
+    ignoreSet(std::move(other.ignoreSet)),
+    isIgnored(std::move(other.isIgnored))
+{
+  // Clean the other object to prevent to objects pointing
+  // at the memory location.
+  if (other.matGram)
+    delete other.matGram;
```

Comment:
Is it necessary to delete matGram. I think we are moving not copying. So, objects can point at the memory location I am not sure. I can be wrong just asking.

Reply:
It's not necessary, I added it as a safeguard. Say we call a destructors of the two objects which point to the same location. They will try to clear the memory at the same location.